server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -499,6 +518,22 @@ public Repository repository(String repositoryName) {
         throw new RepositoryMissingException(repositoryName);
     }
 
+    /**
+     * Returns registered system repository
```

Comment:
Why do we need a separate API? Shouldn't the name still be unique and retrievable via `Repository repository(String repositoryName)` above?

Reply:
this is to make sure we always use `system repository` for remote store .

Reply:
This design is inconsistent, the returned `Repository` exposed the 'isSystem' flag so this information is available to callers if they want to pick and choose.